builtin/common/serialize.lua

```diff
@@ -201,8 +215,8 @@ function core.deserialize(str, safe)
 	local func, err = loadstring(str)
 	if not func then return nil, err end
 
-	-- math.huge is serialized to inf, NaNs are serialized to nan by Lua
-	local env = {inf = math_huge, nan = nan}
+	-- math.huge is serialized to inf, NaNs are serialized to nan by Lua in Minetest 5.6 and earlier
+	local env = {inf = math_huge, nan = 0/0}
```

Comment:
MT 5.5 did serialize nan and inf properly, since that's what `%g` does:

https://github.com/minetest/minetest/commit/5355cb1d87f60bc5548a8a9311e597b9c6046013

And yes, we should bother. Infinities in particular can't be "degraded" gracefully since they might be used as table keys. Using infinities is also reasonable in some circumstances.

Reply:
Until 5.6 the deserialization function has never defined `nan` or `inf` so while they were serialized "correctly" they were never recoverable.

> Infinities in particular can't be "degraded" gracefully since they might be used as table keys.

I know, yet this is what the old code did.

Reply:
> I know, yet this is what the old code did.

And IMO that was a clear bug / shortcoming.